include/AutomatableModel.h

```diff
@@ -80,6 +82,32 @@ class LMMS_EXPORT AutomatableModel : public Model, public JournallingObject
 
 	virtual ~AutomatableModel();
 
+	// Implement those by using the MODEL_IS_VISITABLE macro
+	virtual void accept(ModelVisitor& v) = 0;
+	virtual void accept(ConstModelVisitor& v) const = 0;
+	// use this to make subclasses visitable
+#define MODEL_IS_VISITABLE \
+	void accept(ModelVisitor& v) override { v.visit(*this); } \
+	void accept(ConstModelVisitor& v) const override { v.visit(*this); }
+
+public:
+	//! Return this class casted to Target, or nullptr if impossible
+	template<class Target>
+	Target* dcast(bool doThrow = false)
+	{
+		DCastVisitor<Target> vis; accept(vis);
+		if(doThrow && !vis.result) Q_ASSERT(false);
```

Comment:
Could you provide an example where the cast fails?
Also, please use `if (` instead of `if(`.

Reply:
For example, you have an `AutomatableModel*` pointing to a `ComboBoxModel` or `IntModel`, but try converting the pointer to `FloatModel*` (using `FloatModel` for the `Target` parameter). Should I add this to the comment?

Notes to self: Document `doThrow`, and correct the current statement that `nullptr` is only returned if `!doThrow`. And add something like `see non const version`.